service/integration/open_pbs/README.rst

```diff
@@ -28,57 +36,250 @@ correct the power limit setting if it was not restored as part of the previous
 job. More specifically, if a new power limit is not requested in the next job
 involving the affected nodes (and the save file is present), the hook will
 take the opportunity to restore the power limit setting during the prologue so
-the new job is not affected by a prior power limit setting.
+the new job is not affected by a prior power limit setting. The save file is
+removed when the power limits are restored (regardless of whether the prologue
+or epilogue handles the restore operation).
+
+Automatic Power Limits
+----------------------
+The goal of the automatic power limiter is to provide additional scheduling
+flexibility when the cluster is power constrained. If the cluster power cap is
+sufficiently low, then jobs may not begin executing even when sufficient
+non-power resources are available on idle nodes. At a high level,
+this automatic limiter attempts to set the lowest acceptable power cap to
+each job (based on modeled power-performance relationships and a desired
+slowdown limit) until the scheduler offers to allocate resources to a job. When
+the scheduler offers to start the job, the automatic limiter opportunistically
+increases power to that job if there is additional unallocated power available
+in the cluster.
+
+Automatic power limits are selected through handlers for ``queuejob`` and
+``runjob`` PBS events. The ``queuejob`` event handler initially sets the
+``geopm-job-power-limit`` of each submitted job to be the lowest tolerable
+power limit for that job, according to the configured maximum job slowdown and
+the job's selected power-performance model. Once the PBS
+scheduler has determined that there are sufficient resources (power and
+otherwise) to execute the job, the ``runjob`` event handler is triggered. That
+event handler opportunistically increases the power limit for that job, up to
+the configured maximum power per node (times the number of nodes in the job),
+bound by the amount of power that is still unallocated in the cluster.
+
+Both the ``queuejob`` and ``runjob`` event handlers reconcile differences
+between the automatically configured ``geopm-job-power-limit`` and the
+user-selectable ``geopm-node-power-limit``. In the event that both limits are
+specified, the greater of the two is applied. For example, in a 2-node job with
```

Comment:
pehaps provide a justification for using the greater of the two rather than the lesser of the two?